src/SFA.DAS.Reservations.Application.UnitTests/AccountReservation/Commands/BulkCreateReservationsWithNonLevy/WhenCreatingNewReservations.cs

```diff
@@ -0,0 +1,106 @@
+
+            _command = fixture.Create<BulkCreateReservationsWithNonLevyCommand>();
+
+            _mediator.Setup(x => x.Send(It.IsAny<BulkCreateAccountReservationsCommand>(), _cancellationToken)).ReturnsAsync(_levyResult);
+            _mediator.Setup(x => x.Send(It.IsAny<CreateAccountReservationCommand>(), _cancellationToken)).ReturnsAsync(_nonLevyResult);
+
+            _accountLegalEntitiesService = new Mock<IAccountLegalEntitiesService>();
+            _accountLegalEntitiesService.Setup(x => x.GetAccountLegalEntity(It.IsAny<long>())).ReturnsAsync((long accountLegalEntityId) =>
+            {
+                var accountLegalEntity = fixture.Build<AccountLegalEntity>().
+                         With(x => x.AccountLegalEntityId, accountLegalEntityId).Create();
+                return accountLegalEntity;
+            });
+
+            _mediator.Setup(x => x.Send(It.IsAny<BulkValidateCommand>(), It.IsAny<CancellationToken>())).ReturnsAsync(() => new BulkValidationResults());
+
+            _handler = new BulkCreateReservationsWithNonLevyCommandHandler(_mediator.Object, _accountLegalEntitiesService.Object);
+        }
+
+        [Test]
+        public async Task Then_Levy_Reservations_Are_Created()
+        {
+            //Arrange
+            _command.Reservations.ForEach(x => { x.AccountLegalEntityId = 1; x.TransferSenderAccountId = null; });
+            _accountLegalEntitiesService.Setup(x => x.GetAccountLegalEntity(It.IsAny<long>())).ReturnsAsync((long accountLegalEntityId) =>
+                                    new AccountLegalEntity(Guid.NewGuid(), 1, "account legal entity name", 2, accountLegalEntityId, true, true));
+
+            //Act
+            var result = await _handler.Handle(_command, _cancellationToken);
+
+            //Assert
+            _mediator.Verify(x => x.Send(It.Is<BulkCreateAccountReservationsCommand>(y => y.AccountLegalEntityId == 1 && y.TransferSenderAccountId == null), _cancellationToken), Times.Once);
+            _command.Reservations.All(x => result.BulkCreateResults.Single(y => y.ULN == x.ULN && y.ReservationId != Guid.Empty) != null);
+        }
+
+
+        [Test]
+        public async Task Then_Non_Levy_Reservations_Are_Created()
+        {
+            //Arrange
+            _nonLevyResult.Rule = null;
+            _nonLevyResult.AgreementSigned = true;
+
+            _accountLegalEntitiesService.Setup(x => x.GetAccountLegalEntity(It.IsAny<long>())).ReturnsAsync((long accountLegalEntityId) =>
+                                     new AccountLegalEntity(Guid.NewGuid(), 1, "account legal entity name", 2, accountLegalEntityId, true, false));
+
+            //Act
+            BulkCreateReservationsWithNonLevyResult result = await _handler.Handle(_command, _cancellationToken);
+
+            //Assert
+            _command.Reservations.ForEach(nonLevyEntity =>
+            {
+                _mediator.Verify(x => x.Send(It.Is<CreateAccountReservationCommand>(y =>
```

Comment:
when verifying a number of different properties within a mock call I prefer to use a callback and then inspect the object and Assert the value of the properties

Reply:
It will be difficult to do with the loop. Will need to put more logic around that.